src/gccrs/mod.rs

```diff
@@ -73,4 +77,40 @@ impl Gccrs {
         Gccrs::dump_config()?;
         GccrsConfig::display()
     }
+
+    fn spawn_with_args(args: &Vec<String>) -> Result<ExitStatus> {
+        Command::new("gccrs").args(args).status()
+    }
+
+    fn compile(args: &Vec<String>) -> Result {
+        let mut gccrs_args = GccrsArgs::from_rustc_args(args);
+
+        gccrs_args
+            .drain(..)
```

Comment:
Not sure why you use `drain` that often.

Reply:
Here, `drain` is necessary since we want to use `into_args` on each `GccrsArgs`. But I agree that I was overzealous with the drains, I've removed the one you mentionned previously

Reply:
Yes, but you already own `gccrs_args`, so you should be able to just call `into_args` on the elements of it.

I just tested it and replacing this with 
```suggestion
            .into_iter()
```
also works and is what you normally do.